client/service-container-chain/src/service.rs

```diff
@@ -322,13 +314,23 @@ fn start_consensus_container(
     force_authoring: bool,
     relay_chain_slot_duration: Duration,
     para_id: ParaId,
-    orchestrator_para_id: ParaId,
-    collator_key: CollatorPair,
     overseer_handle: OverseerHandle,
     announce_block: Arc<dyn Fn(Hash, Option<Vec<u8>>) + Send + Sync>,
 ) {
-    let slot_duration = cumulus_client_consensus_aura::slot_duration(&*orchestrator_client)
-        .expect("start_consensus_container: slot duration should exist");
+    let crate::spawner::CollationParams {
+        collator_key,
+        orchestrator_tx_pool,
+        orchestrator_client,
+        orchestrator_para_id,
+        solochain,
+    } = collation_params;
+    let slot_duration = if solochain {
+        // Solochains use Babe instead of Aura, which has 6s slot duration
+        SlotDuration::from_millis(6_000)
```

Comment:
Better to read this from babe configuration then hard code it.

Reply:
can we read it from config? I am not sure if there is a runtime-api from which we can read it

Reply:
Yes. There is a runtime-api. Example: https://github.com/paritytech/polkadot-sdk/blob/f0e420a031b2b0d4c2d3c065ab3e4c43a160b0d7/substrate/bin/node/runtime/src/lib.rs#L2860

Reply:
Actually we already have a variable here called `relay_chain_slot_duration`, I will use that instead. That's also hardcoded to 6s, I will change it to use the runtime api.

Reply:
I didn't find a way to get that using the overseer. We could call the runtime-api directly using the new method, but it seems too much effort, and I need to import sp_consensus_babe to decode the result.

So I'm keeping it hardcoded to 6s.

Leaving the code here for future reference

```rust
let relay_chain_slot_duration = {
    let phash = relay_chain_interface.finalized_block_hash().await.unwrap();
    let babe_config = relay_chain_interface.call_remote_runtime_function_encoded("BabeApi_configuration", phash, &[]).await.unwrap();
    let babe_config: sp_consensus_babe::BabeConfiguration = Decode::decode(&mut babe_config.as_slice()).unwrap();
    
    babe_config.slot_duration
}
```

Reply:
*disregard*